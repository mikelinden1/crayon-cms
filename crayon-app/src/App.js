import React from 'react';
import { Switch, Route } from 'react-router-dom'

import config from 'config';

import Header from 'components/header';
import Homepage from 'components/homepage';
import Login from 'components/login';
import ModuleRoot from 'components/module-root';

class App extends React.Component {
    render() {
        return (
            <div>
                <Header />
                <div className="app-wrapper">
                    <Switch>
                        <Route exact path='/' render={(props) => <Homepage />} />
                        <Route exact path='/login' render={(props) => <Login />} />
                        {config.moduleOrder.map((moduleId) => {
                            return <Route key={`module-route-${moduleId}`} exact path={`/${moduleId}`} render={(props) => <ModuleRoot moduleId={moduleId} {...props} />} />;
                        })}
                    </Switch>

                    <div className="powered-by text-center"><p><em>Powered by Crayon CMS by Mike Linden</em></p></div>
                </div>
            </div>
        );
    }
}

export default App;
