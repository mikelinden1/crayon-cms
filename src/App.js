import React from 'react';

import Header from 'components/header';
import Main from 'components/main';

class App extends React.Component {
    render() {
        return (
            <div>
                <Header />
                <div className="app-wrapper">
                    <Main />
                </div>
            </div>
        );
    }
}

export default App;
