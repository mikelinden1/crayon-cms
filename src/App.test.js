import App from './App';

it('App matches snapshot', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
});