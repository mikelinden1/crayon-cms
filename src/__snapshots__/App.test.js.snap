// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div>
  <Connect(Header) />
  <div
    className="app-wrapper"
  >
    <withRouter(Connect(Main)) />
  </div>
  <div
    className="powered-by text-center"
  >
    <p>
      <em>
        Powered by Crayon CMS by Mike Linden
      </em>
    </p>
  </div>
</div>
`;
